onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1159,7 +1166,11 @@ std::unique_lock<OrtMutex> TensorrtExecutionProvider::GetEngineBuildLock() const
 
 common::Status TensorrtExecutionProvider::Compile(const std::vector<Node*>& fused_nodes,
                                                   std::vector<NodeComputeInfo>& node_compute_funcs) {
-  for (const auto* fused_node : fused_nodes) {
+  size_t fused_nodes_size = fused_nodes.size();
+  std::vector<std::unique_ptr<cudaGraphExec_t>> executable_cuda_graphs(fused_nodes_size);
+  for (size_t node_idx = 0; node_idx < fused_nodes_size; node_idx++) {
+    executable_cuda_graphs[node_idx] = std::make_unique<cudaGraphExec_t>();
```

Comment:
since cudaGraph_t and cudaGraphExec_t are already pointers, i think we can't use make_unique/unique pointers like this ? 
need to see if there's a better way to manage memory of the cudaGraphExec and cudaGraph